File: Examples/Features/Workbook/SaveWorkbookWithPassword.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Workbook
{
    public class SaveWorkbookWithPassword : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            //Change the path to real export path when save.
            XlsxSaveOptions options = new XlsxSaveOptions();
            options.Password = "123456";

            workbook.Save(this.CurrentDirectory + "dest.xlsx", options);

        }

        public override bool CanDownload
        {
            get
            {
                return false;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }
    }
}

File: Examples.VB/Features/Workbook/SaveWorkbookWithPassword.vb
Namespace Features.Workbook
    Public Class SaveWorkbookWithPassword
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Change the path to real export path when save.
            Dim options As New XlsxSaveOptions With {
                .Password = "123456"
            }
            workbook.Save(IO.Path.Combine(CurrentDirectory, "dest.xlsx"), options)
        End Sub
        Public Overrides ReadOnly Property CanDownload As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property IsUpdate As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
